### PR TITLE
fix(cdn): add metric generation frequency in cdn configuration

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/cdn/cdn.go [hunk 1]
```diff
@@ -84,6 +84,9 @@ func (s *Service) ApplyConfiguration(config interface{}) error {
 	if s.Cfg.TCP.GlobalTCPRateLimit == 0 {
 		s.Cfg.TCP.GlobalTCPRateLimit = defaultGlobalTCPRateLimit
 	}
+	if s.Cfg.Metric.Period <= 0 {
+		s.Cfg.Metric.Period = 30
+	}
```
Comment (yesnault) [inline engine/cdn/cdn.go:89]:
```suggestion
	if s.Cfg.Metrics.Period <= 0 {
		s.Cfg.Metrics.Period = 30
	}
```

File: engine/cdn/cdn_metrics.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func (s *Service) initMetrics(ctx context.Context) error {
 func (s *Service) ComputeMetrics(ctx context.Context) {
 	tickPercentil := time.NewTicker(1 * time.Hour)
 	defer tickPercentil.Stop()
-	tickStatsItems := time.NewTicker(30 * time.Second)
+	tickStatsItems := time.NewTicker(time.Duration(s.Cfg.Metric.Period) * time.Second)
```
Comment (yesnault) [inline engine/cdn/cdn_metrics.go:93]:
```suggestion
	tickStatsItems := time.NewTicker(time.Duration(s.Cfg.Metrics.Period) * time.Second)
```

File: engine/cdn/types.go [hunk 3]
```diff
@@ -83,9 +83,12 @@ type Configuration struct {
...
-	API   service.APIServiceConfiguration `toml:"api" comment:"######################\n CDS API Settings \n######################" json:"api"`
-	Log   storage.LogConfig               `toml:"log" json:"log" comment:"###########################\n Log settings.\n##########################"`
-	Units storage.Configuration           `toml:"storageUnits" json:"storageUnits" mapstructure:"storageUnits" comment:"###########################\n Storage Units settings.\n##########################"`
+	API    service.APIServiceConfiguration `toml:"api" comment:"######################\n CDS API Settings \n######################" json:"api"`
+	Log    storage.LogConfig               `toml:"log" json:"log" comment:"###########################\n Log settings.\n##########################"`
+	Units  storage.Configuration           `toml:"storageUnits" json:"storageUnits" mapstructure:"storageUnits" comment:"###########################\n Storage Units settings.\n##########################"`
+	Metric struct {
+		Period int64 `toml:"period" default:"30" json:"period"`
+	} `toml:"metric" comment:"######################\n CDN Metrics Settings \n######################" json:"metric"`
```
Comment (yesnault) [inline engine/cdn/types.go:91]:
```suggestion
	API    service.APIServiceConfiguration `toml:"api" comment:"######################\n CDS API Settings \n######################" json:"api"`
	Log    storage.LogConfig               `toml:"log" json:"log" comment:"###########################\n Log settings.\n##########################"`
	Units  storage.Configuration           `toml:"storageUnits" json:"storageUnits" mapstructure:"storageUnits" comment:"###########################\n Storage Units settings.\n##########################"`
	Metrics struct {
		Period int64 `toml:"period" default:"30" json:"period"`
	} `toml:"metrics" comment:"######################\n CDN Metrics Settings \n######################" json:"metrics"`
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by richardlt — APPROVED** [review]

File: engine/cdn/types.go [hunk 4]
```diff
@@ -83,9 +83,12 @@ type Configuration struct {
...
 	} `toml:"cache" comment:"######################\n CDN Cache Settings \n######################" json:"cache"`
-	API   service.APIServiceConfiguration `toml:"api" comment:"######################\n CDS API Settings \n######################" json:"api"`
-	Log   storage.LogConfig               `toml:"log" json:"log" comment:"###########################\n Log settings.\n##########################"`
-	Units storage.Configuration           `toml:"storageUnits" json:"storageUnits" mapstructure:"storageUnits" comment:"###########################\n Storage Units settings.\n##########################"`
+	API    service.APIServiceConfiguration `toml:"api" comment:"######################\n CDS API Settings \n######################" json:"api"`
+	Log    storage.LogConfig               `toml:"log" json:"log" comment:"###########################\n Log settings.\n##########################"`
+	Units  storage.Configuration           `toml:"storageUnits" json:"storageUnits" mapstructure:"storageUnits" comment:"###########################\n Storage Units settings.\n##########################"`
+	Metrics struct {
+		Period int64 `toml:"period" default:"30" json:"period"`
```
Comment (yesnault) [inline engine/cdn/types.go:90]:
```suggestion
		Period int64 `toml:"period" default:"30" json:"period" comment:"each 30s, metrics are computed"`
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15369.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15370.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**